snekbox/utils/timed.py

```diff
@@ -1,37 +1,34 @@
 """Calling functions with time limits."""
-import multiprocessing
-from collections.abc import Callable, Iterable, Mapping
-from typing import Any, TypeVar
+import signal
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import TypeVar
 
 _T = TypeVar("_T")
 _V = TypeVar("_V")
 
-__all__ = ("timed",)
+__all__ = ("time_limit",)
 
 
-def timed(
-    func: Callable[[_T], _V],
-    args: Iterable = (),
-    kwds: Mapping[str, Any] | None = None,
-    timeout: float | None = None,
-) -> _V:
+@contextmanager
+def time_limit(timeout: int | None = None) -> Generator[None, None, None]:
     """
-    Call a function with a time limit.
+    Decorator to call a function with a time limit. Uses SIGALRM, requires a UNIX system.
```

Comment:
I don't think implementation details should generally be included in the summary. If you think it's useful to point this out, then I think the notes section in the docstring is more appropriate of a place.

```suggestion
    Decorator to call a function with a time limit.
```